server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -72,6 +74,25 @@ public abstract class AbstractMemTable implements IMemTable {
   /** DeviceId -> chunkGroup(MeasurementId -> chunk) */
   private final Map<IDeviceID, IWritableMemChunkGroup> memTableMap;
 
+  /**
+   * Parameters for slightly small estimate to control memory of {@link #memTableMap}, if consider
+   * {@link RamUsageEstimator} as accurate one.
+   *
+   * <p>todo: implement reflection to be more accurate
+   */
+  private static final long WRITABLE_CHUNK_GROUP_SHALLOW_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(WritableMemChunkGroup.class);
+
+  private static final long ALIGNED_CHUNK_GROUP_SHALLOW_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(AlignedWritableMemChunkGroup.class);
+  private static final long WRITABLE_CHUNK_SHALLOW_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(WritableMemChunk.class);
+  private static final long ALIGNED_CHUNK_SHALLOW_SIZE =
+      RamUsageEstimator.shallowSizeOf(AlignedWritableMemChunk.class);
+  private static final long MAP_ENTRY_SIZE = 40L;
+  private static final long MAP_BASE_SIZE = 128L;
+  private static final long INTEGER_SIZE = 16L;
```

Comment:
It's better to add some annotation to explain these values.

Reply:
Thanks ! And more annotation has been added to explain these values, which are figured by `RamUsageEstimater` based on `HashMap` and can be more accurate with further implementation of reflection.